java-spiffe-core/src/main/java/io/spiffe/workloadapi/JwtSource.java

```diff
@@ -0,0 +1,256 @@
+package io.spiffe.workloadapi;
+
+import io.spiffe.bundle.BundleSource;
+import io.spiffe.bundle.jwtbundle.JwtBundle;
+import io.spiffe.bundle.jwtbundle.JwtBundleSet;
+import io.spiffe.exception.BundleNotFoundException;
+import io.spiffe.exception.JwtSourceException;
+import io.spiffe.exception.JwtSvidException;
+import io.spiffe.exception.SocketEndpointAddressException;
+import io.spiffe.spiffeid.SpiffeId;
+import io.spiffe.spiffeid.TrustDomain;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NonNull;
+import lombok.extern.java.Log;
+import lombok.val;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import io.spiffe.bundle.x509bundle.X509Bundle;
+import io.spiffe.svid.jwtsvid.JwtSvid;
+import io.spiffe.svid.jwtsvid.JwtSvidSource;
+
+import java.io.Closeable;
+import java.time.Duration;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.logging.Level;
+
+import static io.spiffe.workloadapi.internal.ThreadUtils.await;
+
+/**
+ * Represents a source of SPIFFE JWT SVIDs and JWT bundles maintained via the Workload API.
+ */
+@Log
+public class JwtSource implements JwtSvidSource, BundleSource<JwtBundle>, Closeable {
+
+    private static final Duration DEFAULT_TIMEOUT;
+
+    static {
+        DEFAULT_TIMEOUT = Duration.parse(System.getProperty("spiffe.newJwtSource.timeout", "PT0S"));
```

Comment:
nit: Could probably save a couple lines of code by combining the declaration and initialization of `DEFAULT_TIMEOUT` on a single line

Reply:
Combined the declaration and initialization, besides that I extracted the property name to an extra constant to make it more visible..